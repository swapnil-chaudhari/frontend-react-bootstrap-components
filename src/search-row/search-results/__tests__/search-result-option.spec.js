import React from 'react';
import renderShallow from 'render-shallow';
import SearchResultOption from '../search-result-option';

describe('<SearchResultOption> when searching for tracks', () => {
    describe('with selected item track', () => {
        const props = {
            className: 'Search',
            artistName: 'Hi-Fi',
            releaseName: 'Aromati vanili',
            imageUrl: 'https://google.com/images/hi-fi',
            isrc: 'BN0123456789',
            option: { upc: 'BN0123456789_012345678905', trackName: 'Nam ne dano' },
            searchName: 'tracks'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <SearchResultOption { ...props } />
            ).output;
        });

        test('renders component with image and content', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('without image link in result item', () => {
        const props = {
            className: 'Search',
            artistName: 'Hi-Fi',
            releaseName: 'Aromati vanili',
            isrc: 'BN0123456789',
            option: { upc: 'BN0123456789_012345678905', trackName: 'Nam ne dano' },
            searchName: 'tracks'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <SearchResultOption { ...props } />
            ).output;
        });

        test('renders empty image placeholder', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with search query', () => {
        const props = {
            className: 'Search',
            artistName: 'Hi-Fi',
            releaseName: 'Aromati vanili',
            imageUrl: 'https://google.com/images/hi-fi',
            isrc: 'BN0123456789',
            option: { upc: 'BN0123456789_012345678905', trackName: 'Nam ne dano' },
            query: 'fi',
            searchName: 'tracks'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <SearchResultOption { ...props } />
            ).output;
        });

        test('renders selected item with highlighting', () => {
            expect(component).toMatchSnapshot();
        });
    });
});

describe('<SearchResultOption> when searching for artists', () => {
    const props = {
        className: 'Search',
        artistName: 'Hi-Fi',
        releaseName: undefined,
        imageUrl: undefined,
        isrc: undefined,
        option: { artistId: '3456789', artistName: 'Hi-Fi' },
        searchName: 'artists'
    };

    describe('without search query', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <SearchResultOption { ...props } />
            ).output;
        });

        test('renders selected item without highlighting', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('with search query', () => {
        let component;

        beforeAll(() => {
            props.query = 'fi';
            component = renderShallow(
                <SearchResultOption { ...props } />
            ).output;
        });

        test('renders selected item with highlighted query', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
