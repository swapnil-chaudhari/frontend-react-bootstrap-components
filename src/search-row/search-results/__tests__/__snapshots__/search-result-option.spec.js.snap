// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultOption> when searching for artists with search query renders selected item with highlighted query 1`] = `
<div
  className="Search"
  onClick={[Function]}
  onMouseMove={[Function]}
  role="option"
>
  
  <div
    className="SearchResultsItem-isrc"
  >
    
  </div>
  <div
    className="SearchResultsItem-trackTitle"
  >
    
  </div>
  <div
    className="SearchResultsItem-artist"
  >
    <span>
      <span>
        Hi-
      </span>
      <span
        className="Select-option-highlighted"
      >
        Fi
      </span>
    </span>
    <span>
      <span>
        
      </span>
    </span>
     
    
  </div>
</div>
`;

exports[`<SearchResultOption> when searching for artists without search query renders selected item without highlighting 1`] = `
<div
  className="Search"
  onClick={[Function]}
  onMouseMove={[Function]}
  role="option"
>
  
  <div
    className="SearchResultsItem-isrc"
  >
    
  </div>
  <div
    className="SearchResultsItem-trackTitle"
  >
    
  </div>
  <div
    className="SearchResultsItem-artist"
  >
    Hi-Fi
     
    
  </div>
</div>
`;

exports[`<SearchResultOption> when searching for tracks with search query renders selected item with highlighting 1`] = `
<div
  className="Search"
  onClick={[Function]}
  onMouseMove={[Function]}
  role="option"
>
  <div
    className="SearchResultsItem-image"
  >
    <img
      className="TrackAlbumArt"
      role="presentation"
      src="https://google.com/images/hi-fi"
    />
  </div>
  <div
    className="SearchResultsItem-isrc"
  >
    BN0123456789
  </div>
  <div
    className="SearchResultsItem-trackTitle"
  >
    Nam ne dano
  </div>
  <div
    className="SearchResultsItem-artist"
  >
    <span>
      <span>
        Hi-
      </span>
      <span
        className="Select-option-highlighted"
      >
        Fi
      </span>
    </span>
    <span>
      <span>
        
      </span>
    </span>
     
    <span>
       · 
      Aromati vanili
    </span>
  </div>
</div>
`;

exports[`<SearchResultOption> when searching for tracks with selected item track renders component with image and content 1`] = `
<div
  className="Search"
  onClick={[Function]}
  onMouseMove={[Function]}
  role="option"
>
  <div
    className="SearchResultsItem-image"
  >
    <img
      className="TrackAlbumArt"
      role="presentation"
      src="https://google.com/images/hi-fi"
    />
  </div>
  <div
    className="SearchResultsItem-isrc"
  >
    BN0123456789
  </div>
  <div
    className="SearchResultsItem-trackTitle"
  >
    Nam ne dano
  </div>
  <div
    className="SearchResultsItem-artist"
  >
    Hi-Fi
     
    <span>
       · 
      Aromati vanili
    </span>
  </div>
</div>
`;

exports[`<SearchResultOption> when searching for tracks without image link in result item renders empty image placeholder 1`] = `
<div
  className="Search"
  onClick={[Function]}
  onMouseMove={[Function]}
  role="option"
>
  <div
    className="SearchResultsItem-image"
  >
    <div
      className="SearchResultsItem-image-placeholder"
    />
  </div>
  <div
    className="SearchResultsItem-isrc"
  >
    BN0123456789
  </div>
  <div
    className="SearchResultsItem-trackTitle"
  >
    Nam ne dano
  </div>
  <div
    className="SearchResultsItem-artist"
  >
    Hi-Fi
     
    <span>
       · 
      Aromati vanili
    </span>
  </div>
</div>
`;
